### PR TITLE
Cherry-pick: Forcibly stop running containers before removal

### PR DESCRIPTION
**Pull request by dcantah**
Signed-off-by: Abhishek Kulkarni <abd.kulkarni@gmail.com>

### CODE REVIEW

**Comment by dcantah** [comment]
Cherry-pick from upstream that makes cri follow the behavior of terminating the sandbox and all of the workload containers when issuing a RemovePodSandbox call. 

https://github.com/containerd/cri/pull/1525/files 
https://github.com/containerd/cri/commit/287c52d1c67921dc29c8ef5a5a5d0e2ba722bb75

**Comment by katiewasnothere** [comment]
Have you run our tests with this? I'm a little concerned about this. I know sometimes we see instances where stopping a container fails or times out and I'm worried that would prevent us from being able to clean up the sandbox.

**Comment by dcantah** [comment]
Our tests call stop first so nothing would change about the behavior but I'll run them. Still confused on how SF was apparently calling only RemovePodSandbox and this was somehow working though..

**Comment by dcantah** [comment]
@katiewasnothere @ambarve @kevpar

**Comment by kevpar** [comment]
Please use `-xs` with `git cherry-pick` so that we get the line telling us the original commit, and your sign-off message.

File: pkg/server/sandbox_remove.go [hunk 1]
```diff
@@ -49,7 +50,10 @@ func (c *criService) RemovePodSandbox(ctx context.Context, r *runtime.RemovePodS
 	// Return error if sandbox container is still running or unknown.
```
Comment (kevpar) [inline pkg/server/sandbox_remove.go]: Can you add another commit to fix this comment?

Comment (dcantah) [reply pkg/server/sandbox_remove.go]: Sure, surprised this is still in master. 
https://github.com/containerd/cri/blob/a0b3b4e4da2e7802927f67efc77feb5433882b0e/pkg/server/sandbox_remove.go#L49

**Comment by dcantah** [comment]
> Please use `-xs` with `git cherry-pick` so that we get the line telling us the original commit, and your sign-off message.

Done

File: pkg/server/sandbox_remove.go (hunk 1 above)
Comment (dcantah) [reply pkg/server/sandbox_remove.go]: Done

**Review by kevpar — APPROVED** [review]
LGTM. Glad we're taking this change.

**Pull request merged**